File: Golang/majority-element.go
func majorityElement(nums []int) int {
    
    n := len(nums)
    m := make(map[int]int)

    for _, val := range nums {
        m[val]++
        if m[val] > n/2 {
            return val
        }
    }
    return -1
}

File: C++/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int majority = 0, counter = 0;

        for (int i = 0; i < nums.size(); i++) {

            if (counter == 0) majority = nums[i];
            nums[i] == majority ? counter++ : counter--;
        }
        return majority;
    }
};